### PR TITLE
Python `CMakeExtension`

### PR DESCRIPTION
**Pull request by rusty1s**
Allows `cmake` build via `pip install .`

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,8 +1,59 @@
...
+
+        extdir = os.path.abspath(osp.dirname(self.get_ext_fullpath(ext.name)))
+
+        if self.debug is None:
+            self.debug = bool(int(os.environ.get('DEBUG', 0)))
+
+        if not osp.exists(self.build_temp):
+            os.makedirs(self.build_temp)
+
+        cmake_args = [
```
Comment (ZenoTan) [inline setup.py:38]: Do we need to disable `BUILD_TEST` for just `pip install`

File: setup.py [hunk 2]
```diff
@@ -1,8 +1,59 @@
...
+            '-DUSE_PYTHON=ON',
+            f'-DCMAKE_LIBRARY_OUTPUT_DIRECTORY={extdir}',
+            f'-DCMAKE_BUILD_TYPE={"DEBUG" if self.debug else "RELEASE"}',
+            f'-DCMAKE_PREFIX_PATH={torch.utils.cmake_prefix_path}',
+            f'-DWITH_CUDA={"ON" if torch.cuda.is_available() else "OFF"}',
```
Comment (ZenoTan) [inline setup.py:43]:
```suggestion
            f'-DWITH_CUDA={"ON" if torch.cuda.is_available() else "OFF"}',
            '-DBUILD_TEST=OFF',
```

File: setup.py [hunk 3]
```diff
@@ -36,5 +87,7 @@
         'dev': dev_requires,
     },
     packages=find_packages(),
+    ext_modules=[CMakeExtension('libpyg')],
```
Comment (ZenoTan) [inline setup.py:90]: A little confused about naming it `libpyg` but in python having `torch.ops.pyg`, and folder name `pyg_lib`. Is there a way to make them look consistent?

File: setup.py (hunk 1 above)
Comment (rusty1s) [reply setup.py:38]: Yes, thank you :)

File: setup.py (hunk 3 above)
Comment (rusty1s) [reply setup.py:90]: I currently do not see an easy way to do this.
* `lib*.so` is the standard naming of libraries
* `pyg` is the namespace
* `pyg_lib` is the Python package name.

I added a TODO here which we can look into later. Hope that works for you :)

**Pull request merged**